src/torchpme/calculators/p3m.py

```diff
@@ -68,13 +62,23 @@ def __init__(
             potential=potential,
             full_neighbor_list=full_neighbor_list,
             prefactor=prefactor,
-            dtype=dtype,
-            device=device,
         )
 
+        if potential.smearing is None:
+            raise ValueError(
+                "Must specify smearing to use a potential with P3MCalculator"
+            )
```

Comment:
I think there is no test for it. Should be go into `workflow` tests.

Also I think more and more that we need an `Ewald` Base class and another "Base" class for the direct calculator.
I fact the EwaldCalculator can serve as a base class and PME and P3M only override the k-space method. But this is something for later...